### PR TITLE
fix: added tests and redactPrivateRepoComments

### PR DESCRIPTION
**Pull request by sshivaditya2019**
Resolves #23 

- Adds tests for `Issue-Deduplication`
- Adds tests for `Task matching`
- Adds redactPrivateRepoComments in the configuration

### CODE REVIEW

**Comment by Keyrxng** [comment]
> @Keyrxng I’ve added some tests for Deduplication and Task Matching that you can pull into your PR. The tests seem to be working well, and the CI is passing with them.

That's awesome dude taking initiative like that, love to see it. Is this ready for review I guess?

File: tests/__sample__/match_threshold_95_1.json
Comment (0x4007) [inline tests/__sample__/match_threshold_95_1.json:1]: I think it makes sense to test with our real specifications.

Comment (Keyrxng) [reply tests/__sample__/match_threshold_95_1.json:1]: > Perhaps there is a way to "train" the system that all of our issues that exist open now are unique?

https://github.com/ubiquity-os/plugins-wishlist/issues/50#issuecomment-2388377812

I believe using this system would be effective. Docs would contain the right context anchors uniformly, then when we fine-tune/train our own model with these docs we train it around these context anchors specifically although they'll be pretty effective without it imo.

If we only have docs for closed tasks then this already effectively treats open tasks as unique and a spec contains the why and typically the how also, which our docs contain for each task, we'll have improved performance in my opinion.

Comment (0x4007) [reply tests/__sample__/match_threshold_95_1.json:1]: Unique means that the plugin should not flag them as redundant. 

This is a tangible test with a boolean result that we can use to tweak our approach.

**Review by 0x4007 — APPROVED** [review]
This pull actually looks fine whats the status?

**Comment by 0x4007** [comment]
@sshivaditya @sshivaditya2019

**Review by 0x4007 — APPROVED** [review]
In the future, you should separate into two different pulls, one adding tests and one solving the problem because larger pulls obviously take a lot more effort to review.

**Comment by 0x4007** [comment]
@sshivaditya2019 there are merge conflicts now so after you fix those, you can merge this in

**Comment by Keyrxng** [comment]
weird

**Pull request closed without merging**